README.md
#  sqflite_preferences

sqflite preferences is an approach to simulate the sharedpreferences behaveir through sqflite library

https://github.com/ahmedkubur/sqflite_preferences

## Usage
How to use

``` dart

//import the library first
import 'package:sqflite_preferences/sqflite_preferences.dart';
//reference to our single class that manages the database
final dbHelper = DatabaseHelper.instance;

```

to initiate your database
``` dart

 initDatabase() async {
dbHelper.initDatabase('phoneNumber', 'INTEGER');
 }

```
to add new or update bool value by key and value
``` dart
setBoolValue()async{
   await dbHelper.setBool('flag',true);
}

```
to get bool value by key
``` dart

bool boolValue;
getBoolValue() async {
     boolValue = await dbHelper.getBool('flag');
}

```
to add new or update int value by key and value
``` dart

setIntValue()async{
   await dbHelper.setInt('number',1);
}

```
to get int value by key
``` dart

int intValue;
getIntValue()async{
     intValue = await dbHelper.getInt('number');
}

```
to add new or update real value by key and value
``` dart

setIntValue()async{
   await dbHelper.setReal('real',1.555);
}

```
to get real by key
``` dart

int intValue;
getIntValue()async{
     intValue = await dbHelper.getReal('real');
}

```
to add new or update blob value by key and value
``` dart

dynamic response = await get(url);
dynamic bytesFromPict = response.image.bodyBytes;
setBlobValue() async {
  await dbHelper.setBlob('pic',bytesFromPic);
}

```
to get blob value by key
``` dart

dynamic blobValue;
getBlobValue() async {
     blobValue = await dbHelper.getBlob('pic');
}
// Example : Image.memory(blobValue), 

```
to add new or update String value by key and value
``` dart

setStringValue() async {
  await dbHelper.setString('name','someone name');
}

```

to get String value by key
``` dart

String stringValue;
getStringValue() async {
     stringValue = await dbHelper.getString('name');
}

```

to get the number of raws by key
``` dart

rowCount() async {
   await dbHelper.queryRowCount('name');
}

```

to delete a saved data permenatly by key
``` dart

deleteValue() async {
   await dbHelper.delete('name');
}

```

### Example

``` dart
import 'package:dbhelper/dbHelper.dart';
import 'package:flutter/material.dart';

var dbHelper = DatabaseHelper.instance;
var value;
void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
        // This makes the visual density adapt to the platform that you run
        // the app on. For desktop platforms, the controls will be smaller and
        // closer together (more dense) than on mobile platforms.
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      // This call to setState tells the Flutter framework that something has
      // changed in this State, which causes it to rerun the build method below
      // so that the display can reflect the updated values. If we changed
      // _counter without calling setState(), then the build method would not be
      // called again, and so nothing would appear to happen.
      _counter++;
    });
  }

  var textValue;
  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text(widget.title),
      ),
      body: Center(
        // Center is a layout widget. It takes a single child and positions it
        // in the middle of the parent.
        child: Column(
          // Column is also a layout widget. It takes a list of children and
          // arranges them vertically. By default, it sizes itself to fit its
          // children horizontally, and tries to be as tall as its parent.
          //
          // Invoke "debug painting" (press "p" in the console, choose the
          // "Toggle Debug Paint" action from the Flutter Inspector in Android
          // Studio, or the "Toggle Debug Paint" command in Visual Studio Code)
          // to see the wireframe for each widget.
          //
          // Column has various properties to control how it sizes itself and
          // how it positions its children. Here we use mainAxisAlignment to
          // center the children vertically; the main axis here is the vertical
          // axis because Columns are vertical (the cross axis would be
          // horizontal).
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Container(
              padding: EdgeInsets.all(50),
              child: TextField(
                onChanged: (value) async {
                  textValue = value;
                  await dbHelper.setString('value', textValue);
                },
              ),
            ),
            RaisedButton(
              child: Text("Get Value"),
              onPressed: () async {
                value = await dbHelper.getString('value');
              },
            ),
            RaisedButton(
              child: Text("Go to next page"),
              onPressed: () {
                Navigator.of(context).push(
                  MaterialPageRoute(
                      fullscreenDialog: true,
                      builder: (BuildContext ctx) {
                        return Page2();
                      }),
                );
              },
            ),
          ],
        ),
      ),
    );
  }
}

class Page2 extends StatefulWidget {
  @override
  _Page2State createState() => _Page2State();
}

class _Page2State extends State<Page2> {
  @override
  void initState() {
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    // var height = MediaQuery.of(context).size.height;
    var width = MediaQuery.of(context).size.width;
    return Scaffold(
      appBar: AppBar(
          // Here we take the value from the MyHomePage object that was created by
          // the App.build method, and use it to set our appbar title.
          // title: Text(widget.title),
          ),
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: <Widget>[
          Center(
            child: Container(
              padding: EdgeInsets.all(30),
              width: width,
              child: FittedBox(
                child: Text(
                  "$value",
                  style: TextStyle(color: Colors.black, fontSize: 30),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}

```

## Developed and coded by Ahmed Ismail Kubur - A.I Kubur :) .

### Buy me a cup of coffee - BTC Wallet (1LQifVDjgqVTyjZJwixmqEaUk6RyDy7fed).


CHANGELOG.md
## [0.0.1] - 28/12/2020.

* sqflite preferences is an approach to simulate the sharedpreferences behaveir through sqflite library

## [0.0.2] - 28/12/2020.

* added example link

## [0.0.3] - 29/12/2020.

* added example and usage

## [0.0.4] - 29/12/2020.

* Edited readme file 

## [0.0.5] - 29/12/2020.

* Edited readme file 

## [0.0.6] - 29/12/2020.

* Edited readme file 

## [0.0.7] - 29/12/2020.

* Edited yaml file 

## [0.0.8] - 29/12/2020.

* Added example 

## [0.0.9] - 21/02/2020.

* updated pubs files 

example/test/widget_test.dart
// This is a basic Flutter widget test.
//
// To perform an interaction with a widget in your test, use the WidgetTester
// utility that Flutter provides. For example, you can send tap and scroll
// gestures. You can also use WidgetTester to find child widgets in the widget
// tree, read text, and verify that the values of widget properties are correct.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:dbhelper/main.dart';

void main() {
  testWidgets('Counter increments smoke test', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(MyApp());

    // Verify that our counter starts at 0.
    expect(find.text('0'), findsOneWidget);
    expect(find.text('1'), findsNothing);

    // Tap the '+' icon and trigger a frame.
    await tester.tap(find.byIcon(Icons.add));
    await tester.pump();

    // Verify that our counter has incremented.
    expect(find.text('0'), findsNothing);
    expect(find.text('1'), findsOneWidget);
  });
}


test/sqflite_preferences_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:sqflite_preferences/sqflite_preferences.dart';

void main() {
  test('adds one to input values', () {
  });
}


example/lib/dbHelper.dart
// Copy Rights Created by A.I.KUBUR

import 'dart:io';
import 'package:path/path.dart';
import 'package:path_provider/path_provider.dart';
import 'package:sqflite/sqflite.dart';

// change the name of the class [DatabaseHelper] to something you like
// make this a singleton class
class DatabaseHelper {
  static final _databaseVersion = 1;
  DatabaseHelper._privateConstructor();
  static final DatabaseHelper instance = DatabaseHelper._privateConstructor();

// only have a single app-wide reference to the database
  static Database database;

// this opens the database (Or creates it if it doesn't not exist)
  initDatabase(dbName, type) async {
    String databaseName = dbName;
    String dataType = type;
    Directory documentsDirectory = await getApplicationDocumentsDirectory();
    String path = join(documentsDirectory.path, "$dbName.db");
    database = await openDatabase(path, version: _databaseVersion,
        onCreate: (Database db, int version) async {
      await db.execute(
          'CREATE TABLE ${databaseName}1 (${databaseName}_key INTEGER, $databaseName $dataType)');
    });
    print("initiated  database $dbName");
  }

// will insert new int value to the column or update the column.
  Future<dynamic> setBool(tableName, bool boolValue) async {
    int value;
    Directory documentsDirectory = await getApplicationDocumentsDirectory();
    String path = join(documentsDirectory.path, "$tableName.db");
    Database _database = await openDatabase(path, version: _databaseVersion,
        onCreate: (Database db, int version) async {
      await db.execute(
          'CREATE TABLE ${tableName}1 (${tableName}_key INTEGER, $tableName INTEGER)');
    });
    Database db = _database;
    var getValue;

    if (boolValue == true) {
      value = 1;
      getValue = Sqflite.firstIntValue(
          await db.rawQuery('SELECT COUNT(*) FROM ${tableName}1'));

      if (getValue == 0) {
        print("data will be inserted");
        return await db.transaction((txn) async {
          return await txn.rawInsert(
              'INSERT INTO ${tableName}1(${tableName}_key, $tableName) VALUES(?, ?)',
              [1, value]);
        });
      } else {
        print("data will be updated");
        return await db.rawUpdate(
            'UPDATE ${tableName}1 SET ${tableName}_key = ?, $tableName = ?',
            [1, value]);
      }
    } else {
      value = 0;

      getValue = Sqflite.firstIntValue(
          await db.rawQuery('SELECT COUNT(*) FROM ${tableName}1'));

      if (getValue == 0) {
        print("data will be inserted");
        return await db.transaction((txn) async {
          return await txn.rawInsert(
              'INSERT INTO ${tableName}1(${tableName}_key, $tableName) VALUES(?, ?)',
              [1, value]);
        });
      } else {
        print("data will be updated");
        return await db.rawUpdate(
            'UPDATE ${tableName}1 SET ${tableName}_key = ?, $tableName = ?',
            [1, value]);
      }
    }
  }

// will insert new int value to the column or update the column.
  Future<dynamic> setInt(tableName, int value) async {
    Directory documentsDirectory = await getApplicationDocumentsDirectory();
    String path = join(documentsDirectory.path, "$tableName.db");
    Database _database = await openDatabase(path, version: _databaseVersion,
        onCreate: (Database db, int version) async {
      await db.execute(
          'CREATE TABLE ${tableName}1 (${tableName}_key INTEGER, $tableName INTEGER)');
    });
    Database db = _database;
    var getValue;

    getValue = Sqflite.firstIntValue(
        await db.rawQuery('SELECT COUNT(*) FROM ${tableName}1'));

    if (getValue == 0) {
      print("data will be inserted");
      return await db.transaction((txn) async {
        return await txn.rawInsert(
            'INSERT INTO ${tableName}1(${tableName}_key, $tableName) VALUES(?, ?)',
            [1, value]);
      });
    } else {
      print("data will be updated");
      return await db.rawUpdate(
          'UPDATE ${tableName}1 SET ${tableName}_key = ?, $tableName = ?',
          [1, value]);
    }
  }

// will insert new double value to the column or update the column.
  Future<dynamic> setReal(tableName, double value) async {
    Directory documentsDirectory = await getApplicationDocumentsDirectory();
    String path = join(documentsDirectory.path, "$tableName.db");
    Database _database = await openDatabase(path, version: _databaseVersion,
        onCreate: (Database db, int version) async {
      await db.execute(
          'CREATE TABLE ${tableName}1 (${tableName}_key INTEGER, $tableName REAL)');
    });
    Database db = _database;
    var getValue;
    getValue = Sqflite.firstIntValue(
        await db.rawQuery('SELECT COUNT(*) FROM ${tableName}1'));
    if (getValue == 0) {
      print("data will be inserted");
      return await db.transaction((txn) async {
        return await txn.rawInsert(
            'INSERT INTO ${tableName}1(${tableName}_key, $tableName) VALUES(?, ?)',
            [1, value]);
      });
    } else {
      print("data will be updated");
      return await db.rawUpdate(
          'UPDATE ${tableName}1 SET ${tableName}_key = ?, $tableName = ?',
          [1, value]);
    }
  }

// will insert new double value to the column or update the column.
  Future<dynamic> setBlob(tableName, dynamic value) async {
    Directory documentsDirectory = await getApplicationDocumentsDirectory();
    String path = join(documentsDirectory.path, "$tableName.db");
    Database _database = await openDatabase(path, version: _databaseVersion,
        onCreate: (Database db, int version) async {
      await db.execute(
          'CREATE TABLE ${tableName}1 (${tableName}_key INTEGER, $tableName BLOB)');
    });
    Database db = _database;
    var getValue;

    getValue = Sqflite.firstIntValue(
        await db.rawQuery('SELECT COUNT(*) FROM ${tableName}1'));

    if (getValue == 0) {
      print("data will be inserted");
      return await db.transaction((txn) async {
        return await txn.rawInsert(
            'INSERT INTO ${tableName}1(${tableName}_key, $tableName) VALUES(?, ?)',
            [1, value]);
      });
    } else {
      print("data will be updated");
      return await db.rawUpdate(
          'UPDATE ${tableName}1 SET ${tableName}_key = ?, $tableName = ?',
          [1, value]);
    }
  }

// will insert new int value to the column or update the column.
  Future<dynamic> setString(dbName, String value) async {
    Directory documentsDirectory = await getApplicationDocumentsDirectory();
    String path = join(documentsDirectory.path, "$dbName.db");
    Database _database = await openDatabase(path, version: _databaseVersion,
        onCreate: (Database db, int version) async {
      await db.execute(
          'CREATE TABLE ${dbName}1 (${dbName}_key INTEGER, $dbName TEXT)');
    });
    Database db = _database;

    var getValue;

    getValue = Sqflite.firstIntValue(
        await db.rawQuery('SELECT COUNT(*) FROM ${dbName}1'));

    if (getValue == 0) {
      print("data will be inserted");
      return await db.transaction((txn) async {
        return await txn.rawInsert(
            'INSERT INTO ${dbName}1(${dbName}_key, $dbName) VALUES(?, ?)',
            [1, value]);
      });
    } else {
      print("data will be updated");
      return await db.rawUpdate(
          'UPDATE ${dbName}1 SET ${dbName}_key = ?, $dbName = ?', [1, value]);
    }
  }

// will return the value of the bool
  Future<dynamic> getBool(dbName) async {
    dynamic stringValue;
    bool boolValue;
    List<Map<String, dynamic>> allRows = [{}];
    Directory documentsDirectory = await getApplicationDocumentsDirectory();
    String path = join(documentsDirectory.path, "$dbName.db");
    Database _database = await openDatabase(path, version: _databaseVersion,
        onCreate: (Database db, int version) async {
      await db.execute(
          'CREATE TABLE ${dbName}1 (${dbName}_key INTEGER, $dbName INTEGER)');
    });
    Database db = _database;
    var getValue;

    getValue = Sqflite.firstIntValue(
        await db.rawQuery('SELECT COUNT(*) FROM ${dbName}1'));
    try {
      allRows = await db.query(dbName + '1');
      stringValue = allRows[getValue - 1][dbName];
      if (stringValue == 1) {
        boolValue = true;
        print("$dbName value is $boolValue");
        return boolValue;
      } else {
        boolValue = false;
        print("$dbName value is $boolValue");
        return boolValue;
      }
    } catch (exception) {
      print("no data was found in $dbName");
      return null;
    }
  }

// will return the value of the int
  Future<dynamic> getInt(dbName) async {
    dynamic intValue;
    List<Map<String, dynamic>> allRows = [{}];
    Directory documentsDirectory = await getApplicationDocumentsDirectory();
    String path = join(documentsDirectory.path, "$dbName.db");
    Database _database = await openDatabase(path, version: _databaseVersion,
        onCreate: (Database db, int version) async {
      await db.execute(
          'CREATE TABLE ${dbName}1 (${dbName}_key INTEGER, $dbName INTEGER)');
    });
    Database db = _database;
    var getValue;

    getValue = Sqflite.firstIntValue(
        await db.rawQuery('SELECT COUNT(*) FROM ${dbName}1'));
    try {
      allRows = await db.query(dbName + '1');
      intValue = allRows[getValue - 1][dbName];
      print("$dbName value is $intValue");
      return intValue;
    } catch (exception) {
      print("no data was found in $dbName");
      return null;
    }
  }

// will return the value of the double
  Future<dynamic> getReal(dbName) async {
    dynamic intValue;
    List<Map<String, dynamic>> allRows = [{}];
    Directory documentsDirectory = await getApplicationDocumentsDirectory();
    String path = join(documentsDirectory.path, "$dbName.db");
    Database _database = await openDatabase(path, version: _databaseVersion,
        onCreate: (Database db, int version) async {
      await db.execute(
          'CREATE TABLE ${dbName}1 (${dbName}_key INTEGER, $dbName REAL)');
    });
    Database db = _database;
    var getValue;

    getValue = Sqflite.firstIntValue(
        await db.rawQuery('SELECT COUNT(*) FROM ${dbName}1'));
    try {
      allRows = await db.query(dbName + '1');
      intValue = allRows[getValue - 1][dbName];
      print("$dbName value is $intValue");
      return intValue;
    } catch (exception) {
      print("no data was found in $dbName");
      return null;
    }
  }

// will return the value of the blob
  Future<dynamic> getBlob(dbName) async {
    dynamic intValue;
    List<Map<String, dynamic>> allRows = [{}];
    Directory documentsDirectory = await getApplicationDocumentsDirectory();
    String path = join(documentsDirectory.path, "$dbName.db");
    Database _database = await openDatabase(path, version: _databaseVersion,
        onCreate: (Database db, int version) async {
      await db.execute(
          'CREATE TABLE ${dbName}1 (${dbName}_key INTEGER, $dbName BLOB)');
    });
    Database db = _database;
    var getValue;

    getValue = Sqflite.firstIntValue(
        await db.rawQuery('SELECT COUNT(*) FROM ${dbName}1'));
    try {
      allRows = await db.query(dbName + '1');
      intValue = allRows[getValue - 1][dbName];
      print("$dbName value is $intValue");
      return intValue;
    } catch (exception) {
      print("no data was found in $dbName");
      return null;
    }
  }

// will return the value of the String
  Future<dynamic> getString(dbName) async {
    dynamic stringValue;
    List<Map<String, dynamic>> allRows = [{}];
    Directory documentsDirectory = await getApplicationDocumentsDirectory();
    String path = join(documentsDirectory.path, "$dbName.db");
    Database _database = await openDatabase(path, version: _databaseVersion,
        onCreate: (Database db, int version) async {
      await db.execute(
          'CREATE TABLE ${dbName}1 (${dbName}_key INTEGER, $dbName TEXT)');
    });
    Database db = _database;
    var getValue;

    getValue = Sqflite.firstIntValue(
        await db.rawQuery('SELECT COUNT(*) FROM ${dbName}1'));
    try {
      allRows = await db.query(dbName + '1');
      stringValue = allRows[getValue - 1][dbName];
      print("$dbName value is $stringValue");
      return stringValue;
    } catch (exception) {
      print("no data was found in $dbName");
      return null;
    }
  }

// All of the methods (insert, query, update, delete) can also be done using
// raw SQL commands. This method uses a raw query to give the row count.
  Future<int> queryRowCount(dbName) async {
    Directory documentsDirectory = await getApplicationDocumentsDirectory();
    String path = join(documentsDirectory.path, "$dbName.db");
    Database _database = await openDatabase(path, version: _databaseVersion);
    Database db = _database;
    try {
      var count = Sqflite.firstIntValue(
          await db.rawQuery('SELECT COUNT(*) FROM ${dbName}1'));
      print("number of raws is: $count");
      return count;
    } catch (exception) {
      print('there are no rows created');
      return null;
    }
  }

// Deletes the column specified by the name and value
  Future<int> delete(dbName) async {
    Directory documentsDirectory = await getApplicationDocumentsDirectory();
    String path = join(documentsDirectory.path, "$dbName.db");
    Database _database = await openDatabase(path, version: _databaseVersion);
    Database db = _database;
    try {
      print("data will be deleted");
      return await db
          .delete(dbName + '1', where: '${dbName}_key = ?', whereArgs: [1]);
    } catch (exception) {
      print('no data where found');
      return null;
    }
  }
}

// how to use

// all you have to do is to add these libraries into your pubspec.yaml
// sqflite: <newest version>
// path_provider: <newest version>

// and import from where you saved this dart file
//import 'package:sqflite_preferences/sqflite_preferences.dart';

// reference to our single class that manages the database
// final dbHelper = DatabaseHelper.instance;

//to initiate your database
//  initDatabase() async {
// dbHelper.initDatabase('phoneNumber', 'INTEGER');
//  }

// to add new or update bool value by key and value
// setBoolValue()async{
//    await dbHelper.setBool('flag',true);
// }

// to get bool value by key
// bool boolValue;
// getBoolValue() async {
//      boolValue = await dbHelper.getBool('flag');
// }

// to add new or update int value by key and value
// setIntValue()async{
//    await dbHelper.setInt('number',1);
// }

// to get int value by key
// int intValue;
// getIntValue()async{
//      intValue = await dbHelper.getInt('number');
// }

// to add new or update real value by key and value
//IMPORTENT NOTE: to get any number for examble 7.0 as is and not lose the ".0" and only get 7 use [name1] [name2] as key
//you can only use [name1] [name2] so use them wisely
// setIntValue()async{
//    await dbHelper.setReal('name1',7.0);
// }
//
// setIntValue()async{
//    await dbHelper.setReal('real',1.555);
// }

// to get real by key
// int intValue;
// getIntValue()async{
//      intValue = await dbHelper.getReal('real');
// }

// to add new or update blob value by key and value
// dynamic response = await get(url);
// dynamic bytesFromPict = response.image.bodyBytes;
// setBlobValue() async {
//   await dbHelper.setBlob('pic',bytesFromPic);
// }

// to get blob value by key
// dynamic blobValue;
// getBlobValue() async {
//      blobValue = await dbHelper.getBlob('pic');
// }
// Example : Image.memory(blobValue),

// to add new or update String value by key and value
// setStringValue() async {
//   await dbHelper.setString('name','someone name');
// }

// to get String value by key
// String stringValue;
// getStringValue() async {
//      stringValue = await dbHelper.getString('name');
// }

// to get the number of raws by key
// rowCount() async {
//    await dbHelper.queryRowCount('name');
// }

// to delete a saved data permenatly by key
// deleteValue() async {
//    await dbHelper.delete('name');
// }
// Copy Rights Created by A.I.KUBUR


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:sqflite_preferences/sqflite_preferences.dart';

void main() {
  runApp(MyApp());
}

// reference to our single class that manages the database
var dbHelper = DatabaseHelper.instance;
// global variable that takes the value of the sqflite stored data
var value;

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
        // This makes the visual density adapt to the platform that you run
        // the app on. For desktop platforms, the controls will be smaller and
        // closer together (more dense) than on mobile platforms.
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  var textValue;
  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: Text(widget.title),
      ),
      body: Center(
        // Center is a layout widget. It takes a single child and positions it
        // in the middle of the parent.
        child: Column(
          // Column is also a layout widget. It takes a list of children and
          // arranges them vertically. By default, it sizes itself to fit its
          // children horizontally, and tries to be as tall as its parent.
          //
          // Invoke "debug painting" (press "p" in the console, choose the
          // "Toggle Debug Paint" action from the Flutter Inspector in Android
          // Studio, or the "Toggle Debug Paint" command in Visual Studio Code)
          // to see the wireframe for each widget.
          //
          // Column has various properties to control how it sizes itself and
          // how it positions its children. Here we use mainAxisAlignment to
          // center the children vertically; the main axis here is the vertical
          // axis because Columns are vertical (the cross axis would be
          // horizontal).
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            Container(
              padding: EdgeInsets.all(50),
              child: TextField(
                onChanged: (value) async {
                  textValue = value;
                  await dbHelper.setString('value', textValue);
                },
              ),
            ),
            RaisedButton(
              child: Text("Get Value"),
              onPressed: () async {
                value = await dbHelper.getString('value');
              },
            ),
            RaisedButton(
              child: Text("Go to next page"),
              onPressed: () {
                Navigator.of(context).push(
                  MaterialPageRoute(
                      fullscreenDialog: true,
                      builder: (BuildContext ctx) {
                        return Page2();
                      }),
                );
              },
            ),
          ],
        ),
      ),
    );
  }
}

class Page2 extends StatefulWidget {
  @override
  _Page2State createState() => _Page2State();
}

class _Page2State extends State<Page2> {
  @override
  void initState() {
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    // var height = MediaQuery.of(context).size.height;
    var width = MediaQuery.of(context).size.width;
    return Scaffold(
      appBar: AppBar(
          // Here we take the value from the MyHomePage object that was created by
          // the App.build method, and use it to set our appbar title.
          // title: Text(widget.title),
          ),
      body: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: <Widget>[
          Center(
            child: Container(
              padding: EdgeInsets.all(30),
              width: width,
              child: FittedBox(
                child: Text(
                  "$value",
                  style: TextStyle(color: Colors.black, fontSize: 30),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}
